pkg/controller/podset/podset_controller.go

```diff
@@ -146,49 +146,75 @@ func (r *PodSetReconciler) reconcilePods(ctx context.Context, podSet *orchestrat
 	currentPodCount := len(activePods)
 	desiredPodCount := int(podSet.Spec.PodGroupSize)
 
-	// The current logic for creating missing pods assumes that existing pods have contiguous indices starting from
-	// 0 (i.e., 0, 1, ..., currentPodCount-1). If a pod with a lower index is deleted for some reason,
-	// this logic could attempt to create a pod with an index that already exists, or it won't fill the gap.
-	// TODO:
-	// 1. Identify all indices of existing active pods.
-	// 2. Determine which indices in the range [0, desiredPodCount-1] are missing.
-	// 3. Create pods for the missing indices.
 	if currentPodCount < desiredPodCount {
-		// Create missing pods
-		podsToCreate := desiredPodCount - currentPodCount
-		for i := 0; i < podsToCreate; i++ {
-			podIndex := currentPodCount + i
-			pod, err := r.createPodFromTemplate(podSet, podIndex)
-			if err != nil {
-				return fmt.Errorf("failed to create pod template: %w", err)
+		// Need Create missing pods
+		switch podSet.Spec.RecreateStrategy {
+		case orchestrationv1alpha1.MissingOnlyPodRecreateStrategy:
+			// Create missing pods
+			existingIndices := map[int]struct{}{}
+
+			// sort pods by index to ensure deterministic creation
+			sortPodsByIndex(activePods)
+			for _, pod := range activePods {
+				if idxStr, ok := pod.Labels[constants.PodGroupIndexLabelKey]; ok {
+					if idx, err := strconv.Atoi(idxStr); err == nil {
+						existingIndices[idx] = struct{}{}
+					}
+				}
 			}
 
-			if err := r.Create(ctx, pod); err != nil {
-				if apierrors.IsAlreadyExists(err) {
-					klog.InfoS("Pod already exists, skipping", "pod", pod.Name, "podset", podSet.Name)
-					continue
+			// fill the gaps
+			for i := 0; i < desiredPodCount; i++ {
+				if _, exists := existingIndices[i]; !exists {
+					pod, err := r.createPodFromTemplate(podSet, i)
+					if err != nil {
+						return fmt.Errorf("failed to create pod template: %w", err)
+					}
+					if err := r.Create(ctx, pod); err != nil {
+						return fmt.Errorf("failed to create pod %v: %w", pod.Name, err)
+					}
+					klog.InfoS("Created pod (missing)", "pod", pod.Name, "podset", podSet.Name)
+				}
+			}
+		case orchestrationv1alpha1.FullRecreatePodRecreateStrategy:
+			// Recreate all pods. First, delete existing active pods.
+			if len(activePods) > 0 {
+				klog.InfoS("FullRecreate strategy: deleting active pods", "podset", podSet.Name, "count", len(activePods))
+				for i := range activePods {
+					if err := r.Delete(ctx, &activePods[i]); err != nil {
```

Comment:
shall we record a eventto explain the podset deletion, since users may wonder why their pod got deleted by controller